### ISSUE TITLE
CollectDuplexSeqMetrics plot

### ISSUE THREAD

**Issue by dilmiperera** [fulcrumgenomics/fgbio#389]
Hi Nils and Tim,

We're using the plots from CollectDuplexSeqMetrics to assess the performance of our experiments. I wonder if you could describe how you calculate the down sampling plot. How do you calculate the "Duplexes - Ideal" family sizes?

<img width="593" alt="screen shot 2018-03-13 at 11 58 05 am" src="https://user-images.githubusercontent.com/29555380/37353495-d8666c7a-26b5-11e8-8186-0e16a1d15b86.png">

Thanks,
Dilmi

**Comment by nh13** [comment]
@dilmiperera See plot number four in [CollectDuplexSeqMetrics.R](https://github.com/fulcrumgenomics/fgbio/blob/master/src/main/resources/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.R#L90-L98), which uses the yield metrics output (`.duplex_yield_metrics.txt` extension) produced by [CollectDuplexSeqMetrics](https://github.com/fulcrumgenomics/fgbio/blob/master/src/main/scala/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.scala).  Documentation for the yield metrics can be found [here](http://fulcrumgenomics.github.io/fgbio/metrics/latest/#duplexyieldmetric), and is created [here](https://github.com/fulcrumgenomics/fgbio/blob/master/src/main/scala/com/fulcrumgenomics/umi/CollectDuplexSeqMetrics.scala#L485-L509).  The idea behind the _ideal_ calculation is the in the documentation for the yield metrics, and the implementation be found reviewing the code (hint: see `pDuplexIdeal`).

**Comment by dilmiperera** [comment]
Thanks. Also, when you say DS families, it includes families with the opposite strand missing?

**Comment by ionox0** [comment]
@dilmiperera To me that does sound right

Also @nh13 should `pDuplexIdeal` have this instead for the range of probabilities to satisfy the minAb and minBa requirements?
```
  def pDuplexIdeal(reads: Int): Double = {
    if (reads < this.minAbReads + this.minBaReads) {
      0
    }
    else {
      val minSsReads = math.min(this.minAbReads, this.minBaReads)
      val maxSsReads = math.max(this.minAbReads, this.minBaReads)

      val binom = new BinomialDistribution(reads, 0.5)
      Range.inclusive(minSsReads, reads - maxSsReads)
        .map(successes => binom.probability(successes))
        .sum
    }
  }
```

**Comment by nh13** [comment]
@ionox0 I don't think we want `maxSsReads` since we need `minSsReads` for a success.  Note that assertions in the comment.  I had to stare at it a while to confirm correctness.

**Comment by ionox0** [comment]
Ok maybe I'm misunderstanding. My thinking was that, in the existing code:

If `reads` = 8, and `minAbReads` = 2, and `minBaReads` = 3,

then `minSsReads` = 2, and BinomialDistribution is calculated from `[2, 6]`. 

But if BinomialDistribution is calculated for `6` reads then this means there are 6 reads on Ab strand and only 2 reads on Ba strand, which doesn't fit `minBaReads` criteria. But I may be misunderstanding the assertion requirements.

**Comment by nh13** [comment]
I’ll have to stare at it when not in mobile. Let me take a closer look this weekend try to arrive at an answer from first principles.

**Comment by ionox0** [comment]
My assumption is that the BinomialDistribution is supposed to be calculate for each number of `successes` that would fit both `minAb` and `minBa` requirements, as in:

if `reads` = 8 and
if `minAb` = 2 and
if `minBa` = 3 then

values of `x` that should be considered success are `2, 3, 4, 5`. 

This corresponds to `x` reads on strand with less stringent requirement, and `reads - x` reads on strand with more strict value for minReads.
